src/QMCTools/PyscfSphToCart.py

```diff
@@ -0,0 +1,502 @@
+#!/usr/bin/env python3
+######################################################################################
+## This file is distributed under the University of Illinois/NCSA Open Source License.
+## See LICENSE file in top directory for details.
+##
+## Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+##
+## File developed by: Kevin Gasperich, kgasperich@anl.gov, Argonne National Laboratory
+##
+## File created by:   Kevin Gasperich, kgasperich@anl.gov, Argonne National Laboratory
+#######################################################################################
+
+import h5py
+import numpy as np
+import itertools
+import shutil
+
+def gms_cart_order(l):
+    def sort1(s):
+        # return single xyz string in gamess order
+        return ''.join(sorted((i*s.count(i) for i in 'xyz'), key=lambda x:(-len(x),x)))
+
+    # unsorted, unordered xyz strings
+    sxyz0 = map(''.join, itertools.combinations_with_replacement('xyz', r=l))
+
+    # unsorted, ordered xyz strings
+    sxyz1 = map(sort1, sxyz0)
+
+    # sorted, ordered xyz strings
+    return sorted(sxyz1, key=lambda s: (sorted((-s.count(i) for i in 'xyz')), s))
+
+def pyscf_cart_order(l):
+    return sorted(map(''.join, itertools.combinations_with_replacement('xyz', r=l)))
+
+def pyscf_to_gms_order(l):
+    xg = gms_cart_order(l)
+    xp = pyscf_cart_order(l)
+    m = np.zeros((len(xp),len(xg)))
+    #p2g = [(xp.index(''.join(sorted(xi))),i) for i,xi in enumerate(xg)]
+    for i,xi in enumerate(xg):
+        m[i,xp.index(''.join(sorted(xi)))] = 1
+    return m.T
+
+def debug_pyscf_to_gms_order(l):
+    xg = gms_cart_order(l)
+    xp = pyscf_cart_order(l)
+    print(xg)
+    print(xp)
+    m = np.zeros((len(xp),len(xg)))
+    #p2g = [(xp.index(''.join(sorted(xi))),i) for i,xi in enumerate(xg)]
+    for i,xi in enumerate(xg):
+        m[i,xp.index(''.join(sorted(xi)))] = 1
+    print(m)
+    return m
+
+s2cdict = {
```

Comment:
Could you add some documentation about what the numbers mean in this structure?

Reply:
yes, I'm still moving some factors around, but I plan to add more documentation

Reply:
I added a description there.

It's not my ideal solution (I should just add a function to generate these for arbitrary l/m), but I wanted to get something working first, and this was simple. 
I squared everything so that I could keep an array of values with finite decimal expansions.

I haven't noticed any issues with accumulation of floating-point errors yet, but I should probably consider whether that might become a problem for high enough angular momenta.
(This actually led me to your jupyter notebooks yesterday: I `grep`ped for `factorial` while trying to decide whether to take a product of ratios `prod(a_i/b_i)` or a ratio of products `prod(a_i)/prod(b_i)` and found a comment that mentioned the `GaussianOrbitals` notebook from the qmc_algorithms repo)

Reply:
@prckent @kgasperich 

I tested this and it works and solves the #5205 issue.